.github/workflows/enzyme-rust.yml

```diff
@@ -0,0 +1,98 @@
+name: Enzyme Rust CI
+
+on: 
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+  merge_group:
+
+jobs:
+  rust-autodiff:
+    name: Rust Autodiff ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [linux-x86-n2-32]
+    
+    timeout-minutes: 90
+
+    container:
+      image: ${{ (contains(matrix.os, 'linux') && 'ghcr.io/enzymead/reactant-docker-images@sha256:91e1edb7a7c869d5a70db06e417f22907be0e67ca86641d48adcea221fedc674' ) || '' }}
+    
+    steps:
+    - name: Install dependencies
+      run: |
+        apt-get update
+        apt-get install -y binutils ninja-build cmake gcc g++ python3 python3-dev ccache nodejs npm
+    - name: Install Rust toolchain
+      run: |
+        curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
+        . ~/.cargo/env
+        # Pin to specific nightly commit for reproducibility
+        rustup toolchain install nightly-2024-12-15
+        rustup default nightly-2024-12-15
+    - uses: actions/checkout@v4
+      name: Checkout Enzyme
+    - name: Clone and configure Rust compiler
+      run: |
+        . ~/.cargo/env
+        git clone --depth 1 https://github.com/rust-lang/rust.git
+        cd rust
+        ./configure \
+          --enable-llvm-link-shared \
+          --enable-llvm-plugins \
+          --enable-llvm-enzyme \
+          --release-channel=nightly \
+          --enable-llvm-assertions \
+          --enable-clang \
+          --enable-option-checking \
+          --enable-ninja \
+          --disable-docs \
+          --enable-ccache
+    - name: Setup ccache
+      run: |
+        echo "CCACHE_DIR=$HOME/.ccache" >> $GITHUB_ENV
+        echo "CC=ccache gcc" >> $GITHUB_ENV
+        echo "CXX=ccache g++" >> $GITHUB_ENV
```

Comment:
Who uses this? What exactly is built using GCC? I think Rust still uses LLVM. The rust invocation will prefix its invocations with `ccache` by itself as far as I can see, thanks to the `--enable-ccache` a few lines above.